substrate/client/rpc-spec-v2/src/transaction/transaction_broadcast.rs

```diff
@@ -100,13 +119,31 @@ where
 	<Pool::Block as BlockT>::Hash: Unpin,
 	Client: HeaderBackend<Pool::Block> + BlockchainEvents<Pool::Block> + Send + Sync + 'static,
 {
-	fn broadcast(&self, bytes: Bytes) -> RpcResult<Option<String>> {
+	async fn broadcast(
+		&self,
+		connection_details: ConnectionDetails,
+		bytes: Bytes,
+	) -> RpcResult<Option<String>> {
 		let pool = self.pool.clone();
 
 		// The unique ID of this operation.
 		let id = self.generate_unique_id();
 
-		let mut best_block_import_stream =
+		// Ensure that the connection has not reached the maximum number of active operations.
+		let Some(reserved_connection) = self.rpc_connections.reserve_space(connection_details.id())
+		else {
+			return Ok(None)
+		};
+		let Some(reserved_identifier) = reserved_connection.register(id.clone()) else {
+			// This can only happen if the generated operation ID is not unique.
+			return Ok(None)
+		};
+
+		// The compiler can no longer deduce the type of the stream and complains
+		// about `one type is more general than the other`.
+		let mut best_block_import_stream: std::pin::Pin<
+			Box<dyn Stream<Item = <Pool::Block as BlockT>::Hash> + Send>,
+		> =
```

Comment:
I wonder why this type annotation needed adding!

Reply:
This is an edge-case in the rust compiler, where the rustc cannot deduce the type of this stream.

I'm not entirely sure what is happening behind the scenes, I expect the async block cannot be properly deduced here.

The error I get is:
```rust
error[E0308]: mismatched types
   --> substrate/client/rpc-spec-v2/src/transaction/transaction_broadcast.rs:222:3
    |
146 |                 |notification| async move { notification.is_new_best.then_some(notification.hash) },
    |                                --------------------------------------------------------------------
    |                                |
    |                                the expected `async` block
    |                                the found `async` block
...
222 |         sc_rpc::utils::spawn_subscription_task(&self.executor, fut);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ one type is more general than the other
    |
    = note: expected `async` block `{async block@substrate/client/rpc-spec-v2/src/transaction/transaction_broadcast.rs:146:20: 146:88}`
               found `async` block `{async block@substrate/client/rpc-spec-v2/src/transaction/transaction_broadcast.rs:146:20: 146:88}`
note: the lifetime requirement is introduced here
   --> /home/lexnv/workspace/polkadot-sdk/substrate/client/rpc/src/utils.rs:178:34
    |
178 |     fut: impl Future<Output = ()> + Send + 'static,
    |                                     ^^^^

For more information about this error, try `rustc --explain E0308`.
```

Reply:
Try bumping jsonrpsee to v0.22.4 it has fixed it for me locally